src/PactNet.Native/NativeMockServer.cs

```diff
@@ -43,75 +81,81 @@ public string MockServerMismatches(int mockServerPort)
             IntPtr matchesPtr = Interop.MockServerMismatches(mockServerPort);
 
             return matchesPtr == IntPtr.Zero
-                       ? string.Empty
-                       : Marshal.PtrToStringAnsi(matchesPtr);
+                ? string.Empty
+                : Marshal.PtrToStringAnsi(matchesPtr);
         }
 
         public string MockServerLogs(int mockServerPort)
         {
             IntPtr logsPtr = Interop.MockServerLogs(mockServerPort);
 
             return logsPtr == IntPtr.Zero
-                       ? string.Empty
-                       : Marshal.PtrToStringAnsi(logsPtr);
+                ? string.Empty
+                : Marshal.PtrToStringAnsi(logsPtr);
         }
 
         public bool CleanupMockServer(int mockServerPort)
             => Interop.CleanupMockServer(mockServerPort);
 
         public void WritePactFile(int mockServerPort, string directory, bool overwrite)
+            => VerifyResult(() => Interop.WritePactFile(mockServerPort, directory, overwrite));
+
+        private void VerifyResult(Func<int> func)
         {
-            int result = Interop.WritePactFile(mockServerPort, directory, overwrite);
+            var result = func.Invoke();
 
-            if (result == 0)
+            if (result != 0)
```

Comment:
I think this read better before because there was less nesting.

Reply:
Yes, the rule of thump is to nest the non-optimal case, baring is doesn't complicate your code, in order to clearly see what the method is doing optimally :)